### PR TITLE
Make a pass separating atomic.numbers from numeric

### PR DESCRIPTION
**Pull request by rocky**
Here I've separated out all of the functions from `atomic.numbers` (Representation of Numbers) that aren't listed in https://reference.wolfram.com/language/guide/RepresentationOfNumbers.html

As always, there is more work that could be done, but this is all the time I have for this for now. As as is custom, my take is with the exception of whatever smallish changes noticed, we merge and continue.

There are probably more functions elsewhere that belong here. And a number of the inappropriate ones have been shuffled back to `numeric` (Numerical Functions) or https://reference.wolfram.com/language/guide/NumericalFunctions.html

### CODE REVIEW

**Comment by mmatera** [comment]
LGTM

File: mathics/builtin/numbers/exptrig.py [hunk 1]
```diff
@@ -21,11 +25,140 @@
 from mathics.core.symbols import Symbol
 
 from mathics.builtin.arithmetic import _MPMathFunction
-from mathics.builtin.atomic.numbers import Fold
 
 from mathics.core.attributes import listable, numeric_function, protected
 
 
+class Fold(object):
```
Comment (TiagoCavalcante) [inline mathics/builtin/numbers/exptrig.py:32]: In Python3 the `(object)` part isn't needed, and usually is hidden.

File: mathics/builtin/numbers/exptrig.py [hunk 2]
```diff
@@ -21,11 +25,140 @@
...
+        FLOAT: ComputationFunctions(
+            cos=math.cos,
+            sin=math.sin,
+        ),
+        MPMATH: ComputationFunctions(
+            cos=mpmath.cos,
+            sin=mpmath.sin,
+        ),
+        SYMBOLIC: ComputationFunctions(
+            cos=lambda x: Expression("Cos", x),
```
Comment (TiagoCavalcante) [inline mathics/builtin/numbers/exptrig.py:52]: SymbolCos

File: mathics/builtin/numbers/exptrig.py [hunk 3]
```diff
@@ -21,11 +25,140 @@
...
+            sin=lambda x: Expression("Sin", x),
```
Comment (TiagoCavalcante) [inline mathics/builtin/numbers/exptrig.py:53]: SymbolSin

File: mathics/builtin/string/operations.py [hunk 4]
```diff
@@ -45,6 +47,96 @@
...
+        h = hash_func()
+        user_hash(h.update)
+        res = h.hexdigest()
+        if py_format in ("HexString", "HexStringLittleEndian"):
+            return String(res)
+        res = int(res, 16)
+        if py_format == "DecimalString":
+            return String(str(res))
+        elif py_format == "ByteArray":
+            return from_python(bytearray(res))
```
Comment (TiagoCavalcante) [inline mathics/builtin/string/operations.py:130]: from_python can be replaced by an Expression.

**Review by TiagoCavalcante — APPROVED** [review]
@rocky I know that the changes you've made are related to move the code from one place to another, but I believe it is good to use the standard on every added line.

File: mathics/builtin/numbers/exptrig.py (hunk 1 above)
Comment (mmatera) [reply mathics/builtin/numbers/exptrig.py:32]: BTW, `Fold` is also a Builtin. Maybe this is an opportunity to rename it to `_Fold` or something else...

File: mathics/builtin/string/operations.py (hunk 4 above)
Comment (rocky) [reply mathics/builtin/string/operations.py:130]: @TiagoCavalcante Please just go ahead and make the changes you suggest. Thanks.

**Pull request merged**